### DISCUSSION TITLE
nixos-rebuild switch fails due to incompatible "init"

### DISCUSSION THREAD

**Discussion by vaci** [nix-community/NixOS-WSL#391] [Q&A] [answered]
Hi,

I was able to install the latest NixOS-WSL image as the default WSL2 distribution just fine, but then running 

  wsl -d NixOS

exits immediately, whereas I would expect to drop into a shell.

The only way I can get a shell in the NixOS subsystem is as follows:

 wsl --user root -e /nix/var/nix/profiles/system/sw/bin/bash

Then, I can run nix-channel update just fine, but I cannot run "nixos-rebuild switch" because I get the following error:

```
building the system configuration...
Warning: the new NixOS configuration has an ‘init’ that is
incompatible with the current configuration.  The new configuration
won't take effect until you reboot the system.
warning: error(s) occurred while switching to the new configuration
```

.. which makes sense, as I guess I'm not running systemd as the init process?

I'm new to WSL2 to not sure how to diagnose why the subsystem won't run straight out of the box?

Thanks
Vaci

**Comment by 573** [comment]
See here for a possibly duplicate question pls:

>If you can not switch due to systemd not running, you should still be able to rebuild the systemd by using boot instead. Then just restart the distro or VM with either `wsl -t <distro>` or `wsl --shutdown`
>
>Does NixOS work normally when you try to import the latest release's `nixos-wsl.tar`?

_Originally posted by @nzbr in https://github.com/nix-community/NixOS-WSL/discussions/340#discussioncomment-8027248_

**Reply by vaci** [reply]
I figured that I'm using the version of WSL that is pre-installed on Windows, not the latest WSL that is available on the Microsoft Store, which I suspect might be the issue as the older WSL doesn't support using systemd as init?

**Comment by vaci** [comment, marked as answer]
As I suspected, just upgrading to WSL2 fixed this issue.
